aptos-move/framework/aptos-framework/sources/event.move

```diff
@@ -54,4 +54,18 @@ module aptos_framework::event {
     public fun destroy_handle<T: drop + store>(handle: EventHandle<T>) {
         EventHandle<T> { counter: _, guid: _ } = handle;
     }
+
+    #[test_only]
+    public fun emitted_events<T: drop + store>(handle: &EventHandle<T>): vector<T> {
+        emitted_events_internal(bcs::to_bytes(&handle.guid))
+    }
+
+    #[test_only]
+    public fun was_event_emitted<T: drop + store>(handle: &EventHandle<T>, msg: &T): bool {
```

Comment:
assert_emission?

Reply:
assert is typically tied to an `assert` statement that explodes if false. here we want the user space to be able to dictate the behavior if the event exists or not.